bot/seasons/christmas/adventofcode.py

```diff
@@ -196,14 +198,37 @@ def __init__(self, bot: commands.Bot):
     @adventofcode_group.command(name="join", aliases=("j",), brief="Learn how to join PyDis' private AoC leaderboard")
     async def join_leaderboard(self, ctx: commands.Context):
         """
-        Retrieve the link to join the PyDis AoC private leaderboard
+        DM the user the information for joining the PyDis AoC private leaderboard
         """
 
+        author = ctx.message.author
+        log.info(f"{author.name} ({author.id}) has requested the PyDis AoC leaderboard code")
+
         info_str = (
             "Head over to https://adventofcode.com/leaderboard/private "
-            f"with code `{AocConfig.leaderboard_join_code}` to join the PyDis private leaderboard!"
+            f"with code `{AOC_JOIN_CODE}` to join the PyDis private leaderboard!"
         )
-        await ctx.send(info_str)
+        try:
+            await author.send(info_str)
+        except discord.errors.Forbidden:
+            log.debug(f"{author.name} ({author.id}) has disabled DMs from server members")
+            await ctx.send(
+                f':x: {author.mention}, please (temporarily) enable DMs to receive the join code'
+            )
+
+    @with_role(Roles.admin, Roles.owner)
+    @adventofcode_group.command(name="changecode", hidden=True)
+    async def update_aoc_join_code(self, ctx: commands.Context, new_code: str):
+        """
+        Staff command to update the AoC join code constant locally to allow for the code to be updated
+        on regeneration without having to redeploy the bot
+        """
+
+        author = ctx.message.author
+        log.info(f"{author.name} ({author.id}) has changed the PyDis AoC leaderboard code")
+
+        global AOC_JOIN_CODE  # Necessary (probably?) evil to update the code without redeploying
+        AOC_JOIN_CODE = new_code
```

Comment:
Is there any point in this command? we'll have to update the bot and reboot anyway with the new env so we might as well just do it the proper way.